mlflow/server/js/src/components/modals/RenameRunModal.js

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import RenameRunFormView from './RenameRunFormView';
+
+import Utils from '../../utils/Utils';
+import ReactModal from 'react-modal';
+
+import { setTagApi, getUUID } from '../../Actions';
+import { withRouter } from 'react-router-dom';
+import Routes from "../../Routes";
+
+
+const modalStyles = {
+  content: {
+    top: '50%',
+    left: '50%',
+    right: 'auto',
+    bottom: 'auto',
+    marginRight: '-50%',
+    transform: 'translate(-50%, -50%)'
+  },
+  overlay: {
+    backgroundColor: 'rgba(33, 37, 41, .75)'
+  }
+};
+
+class RenameRunModal extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.onRequestCloseHandler = this.onRequestCloseHandler.bind(this);
+  }
+
+  state = {
+    isSubmittingState: false,
+  }
+
+  static propTypes = {
+    open: PropTypes.bool,
+    experimentId: PropTypes.number.isRequired,
+    runUuid: PropTypes.string.isRequired,
+    runName: PropTypes.string.isRequired,
+    onClose: PropTypes.func.isRequired,
+    dispatch: PropTypes.func.isRequired,
+    history: PropTypes.object.isRequired,
+  }
+
+  /**
+   * Form-submission handler with method signature as prescribed by Formik.
+   * See https://github.com/jaredpalmer/formik#how-form-submission-works for an explanation
+   * of how / when this method is called.
+   */
+  handleSubmit = (
+    values,
+    {
+      setSubmitting,
+    }) => {
+    const { newRunName } = values;
+    this.setState({ isSubmittingState: true });
+    const tagKey = Utils.runNameTag;
+    const setTagRequestId = getUUID();
+    return this.props.dispatch(
+      setTagApi(this.props.runUuid, tagKey, newRunName, setTagRequestId)).catch((err) => {
+        // TODO: remove alert, redirect to an error page on failed requests once one exists
+        // eslint-disable-next-line no-alert
+        alert("Unable to rename run, got error '" + err + "'. Redirecting to parent experiment " +
+        "page.");
+        this.props.history.push(Routes.getExperimentPageRoute(this.props.experimentId));
+      }).finally(() => {
+        this.setState({ isSubmittingState: false });
```

Comment:
It's a bit unfortunate to be duplicating this submitting state with Formik, but if you want to prevent closing the modal while submitting, I don't see a better way.